core/state.py

```diff
@@ -475,6 +475,18 @@ def PopTemp(self):
     self.var_stack.pop()
     #util.log('**** PopTemp()')
 
+  def PushSource(self, argv):
+    """For the scope of a file read with 'source' or '.'."""
+    self.func_name_stack.append("source")
+    self.argv_stack.append(_ArgFrame(argv))
+    # Bash treats 'locals' in the scope of the sourced file as local to the
```

Comment:
Hm I'm confused by this wording.  Are you just referring to top-level variables in the sourced file?

I think this is better illustrated by a spec test.  You can refer to it from a comment in the code.

Reply:
Yes, just top-level variables.